### PR TITLE
SSCS-9222: Select hearing for Upload hearing recording event

### PR DESCRIPTION
**Pull request by divisathyan**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/SSCS-9222

**Does this PR introduce a breaking change?** (check one with "x")
```
[ ] Yes
[ x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/uploadhearingrecording/UploadHearingRecordingAboutToStartHandler.java [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+    private boolean isBeforeDate(LocalDateTime currentDate, String hearingDate, String hearingTime) {
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm");
+        return LocalDateTime.parse(hearingDate + " " + hearingTime, formatter).isBefore(currentDate);
+    }
+
+    @NotNull
+    private String selectHearing(Hearing hearing) {
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd");
+        DateTimeFormatter resultFormatter = DateTimeFormatter.ofPattern("dd MMM yyyy");
```
Comment (ryandac) [inline src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/uploadhearingrecording/UploadHearingRecordingAboutToStartHandler.java:84]: these two variables could be private static final

**Review by ryandac — APPROVED** [review]

File: build.gradle [hunk 2]
```diff
@@ -320,7 +320,7 @@ dependencies {
     compile group: 'uk.gov.hmcts.reform', name: 'doc-assembly-client', version: '1.07'
     compile group: 'uk.gov.hmcts.reform', name: 'document-management-client', version: '7.0.0'
 
-    compile group: 'com.github.hmcts', name: 'sscs-common', version: '4.3.27'
+    compile group: 'com.github.hmcts', name: 'sscs-common', version: '4.3.30'
```
[the remark this answers is not in the document]
Comment (divisathyan) [reply build.gradle:323]: ok

**Review by chrisdavecm — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/uploadhearingrecording/UploadHearingRecordingAboutToStartHandler.java [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+        }
+
+        final CaseDetails<SscsCaseData> caseDetails = callback.getCaseDetails();
+        final SscsCaseData sscsCaseData = caseDetails.getCaseData();
+        PreSubmitCallbackResponse<SscsCaseData> response = new PreSubmitCallbackResponse<>(sscsCaseData);
+
+        if (sscsCaseData.getHearings() == null || sscsCaseData.getHearings().isEmpty()) {
+            response.addError("No hearing has been conducted on this case");
+            return response;
+        } else {
```
[the remark this answers is not in the document]
Comment (divisathyan) [reply src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/uploadhearingrecording/UploadHearingRecordingAboutToStartHandler.java:60]: done

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/uploadhearingrecording/UploadHearingRecordingAboutToStartHandler.java (hunk 1 above)
Comment (divisathyan) [reply src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/uploadhearingrecording/UploadHearingRecordingAboutToStartHandler.java:84]: done

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/uploadhearingrecording/UploadHearingRecordingAboutToStartHandler.java [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+            if (validHearings.isEmpty()) {
+                response.addError("No hearing has been conducted on this case");
+                return response;
+            }
+            sscsCaseData.setSelectHearingDetails(new DynamicList(new DynamicListItem("", ""), validHearings));
+        }
+        return response;
+    }
+
+    private boolean isBeforeDate(LocalDateTime currentDate, String hearingDate, String hearingTime) {
```
[the remark this answers is not in the document]
Comment (divisathyan) [reply src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/uploadhearingrecording/UploadHearingRecordingAboutToStartHandler.java:76]: Removed private method to use DateTimeUtils directly

**Pull request merged**